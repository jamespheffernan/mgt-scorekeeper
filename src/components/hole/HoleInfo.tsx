import React from 'react';
import { useGameStore } from '../../store/gameStore';
import { millbrookDb } from '../../db/millbrookDb';
import { Course, TeeOption } from '../../db/courseModel';

interface HoleInfoProps {
  holeNumber: number;
}

export const HoleInfo: React.FC<HoleInfoProps> = ({ holeNumber }) => {
  const [course, setCourse] = React.useState<Course | null>(null);
  const [teeOptions, setTeeOptions] = React.useState<TeeOption[]>([]);
  const [isLoading, setIsLoading] = React.useState(true);
  
  // Get match info from store
  const match = useGameStore(state => state.match);
  const playerTeeIds = match.playerTeeIds || [];
  
  // Load course and tee information on mount
  React.useEffect(() => {
    const loadCourseInfo = async () => {
      if (!match.courseId) return;
      
      try {
        setIsLoading(true);
        const courseData = await millbrookDb.getCourse(match.courseId);
        
        if (courseData) {
          setCourse(courseData);
          
          // Get unique tee options being used by players
          const uniqueTeeIds = [...new Set(playerTeeIds)];
          const usedTeeOptions = uniqueTeeIds
            .map(teeId => courseData.teeOptions.find(tee => tee.id === teeId))
            .filter(Boolean) as TeeOption[];
            
          setTeeOptions(usedTeeOptions);
        }
      } catch (error) {
        console.error('Error loading course info:', error);
      } finally {
        setIsLoading(false);
      }
    };
    
    loadCourseInfo();
  }, [match.courseId, playerTeeIds]);
  
  // Get hole information for the current hole
  const getHoleInfo = (tee: TeeOption) => {
    if (!tee || !tee.holes) return null;
    
    const holeInfo = tee.holes.find(h => h.number === holeNumber);
    return holeInfo;
  };
  
  if (isLoading) {
    return <div className="hole-info-loading">Loading hole information...</div>;
  }
  
  if (!course || teeOptions.length === 0) {
    return (
      <div className="hole-info">
        <h3>Hole {holeNumber}</h3>
        <p>No course information available</p>
      </div>
    );
  }
  
  return (
    <div className="hole-info">
      <h3>Hole {holeNumber}</h3>
      
      <div className="hole-info-details">
        {teeOptions.map(tee => {
          const holeInfo = getHoleInfo(tee);
          if (!holeInfo) return null;
          
          return (
            <div key={tee.id} className="hole-tee-info">
              <div 
                className="tee-color-badge"
                style={{ backgroundColor: tee.color.toLowerCase() }}
              >
                {tee.name}
              </div>
              <div className="hole-stats">
                <span className="hole-yardage">{holeInfo.yardage} yards</span>
                <span className="hole-par">Par {holeInfo.par}</span>
                <span className="hole-si">SI: {holeInfo.strokeIndex}</span>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
}; 