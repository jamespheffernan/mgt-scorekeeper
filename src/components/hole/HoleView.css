/* Styles for HoleView and HoleViewMobile, extracted from App.css */

.hole-view {
  background-color: #f9f9f9;
  padding: 24px;
  border-radius: 12px;
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.08);
  max-width: 1200px;
  margin: 0 auto;
}

.hole-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 24px;
  border-bottom: 1px solid #e2e8f0;
  padding-bottom: 18px;
}

.hole-header h2 {
  font-size: 32px;
  color: #334155;
  margin: 0;
  font-weight: 700;
  position: relative;
}

.hole-header h2::after {
  content: '';
  position: absolute;
  bottom: -8px;
  left: 0;
  width: 40px;
  height: 3px;
  background-color: #3b82f6;
  border-radius: 2px;
}

.hole-actions {
  display: flex;
  gap: 12px;
}

.view-ledger-button, .finish-round-button, .cancel-game-button {
  padding: 10px 18px;
  border-radius: 8px;
  font-size: 14px;
  font-weight: 600;
  color: white;
  border: none;
  cursor: pointer;
  transition: all 0.2s;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.view-ledger-button {
  background-color: #3b82f6;
}

.view-ledger-button:hover {
  background-color: #2563eb;
  transform: translateY(-2px);
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}

.finish-round-button {
  background-color: #10b981;
}

.finish-round-button:hover {
  background-color: #059669;
  transform: translateY(-2px);
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}

.cancel-game-button {
  background-color: #ef4444;
}

.cancel-game-button:hover {
  background-color: #dc2626;
  transform: translateY(-2px);
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}

.scores-container {
  margin-bottom: 28px;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.05);
  padding: 20px;
  border: 1px solid #edf2f7;
}

.scores-container h3 {
  font-size: 20px;
  margin: 0 0 20px 0;
  color: #2d3748;
  padding-bottom: 12px;
  border-bottom: 1px solid #f0f4f8;
}

.player-score-row {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 16px;
  padding: 16px;
  border-radius: 10px;
  border-left: 4px solid;
  background-color: #f8fafc;
  transition: all 0.2s;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.03);
}

.player-score-row:hover {
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.06);
  transform: translateY(-1px);
}

.player-score-row.Red {
  border-left-color: var(--color-red);
}

.player-score-row.Blue {
  border-left-color: var(--color-blue);
}

.player-team-indicator {
  flex: 1;
}

.score-selector {
  width: 100px;
}

.score-selector select {
  width: 100%;
  padding: 8px 12px;
  border-radius: 6px;
  border: 1px solid #cbd5e1;
  font-size: 14px;
  background-color: white;
  appearance: none;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24' stroke='%2364748b'%3E%3Cpath stroke-linecap='round' stroke-linejoin='round' stroke-width='2' d='M19 9l-7 7-7-7'%3E%3C/path%3E%3C/svg%3E");
  background-repeat: no-repeat;
  background-position: right 8px center;
  background-size: 16px;
  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.05);
  transition: all 0.2s;
}

.score-selector select:hover {
  border-color: #a0aec0;
}

.score-selector select:focus {
  border-color: #3b82f6;
  box-shadow: 0 0 0 2px rgba(59, 130, 246, 0.2);
  outline: none;
}

.junk-flags {
  display: flex;
  flex-wrap: wrap;
  gap: 10px;
  margin-left: 20px;
}

.junk-flag-label {
  display: inline-flex;
  align-items: center;
  gap: 6px;
  padding: 6px 10px;
  border-radius: 6px;
  background-color: #f1f5f9;
  cursor: pointer;
  transition: all 0.2s;
  border: 1px solid transparent;
}

.junk-flag-label:hover {
  background-color: #e2e8f0;
}

.junk-flag-label input {
  margin: 0;
  width: 16px;
  height: 16px;
  accent-color: #3b82f6;
}

.junk-flag-label input:checked + .junk-name {
  font-weight: 600;
  color: #2563eb;
}

.junk-name {
  font-size: 13px;
  font-weight: 500;
  color: #4b5563;
  transition: all 0.2s;
}

.status-bar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: var(--color-light);
  padding: 12px;
  border-radius: var(--border-radius);
  margin-bottom: 24px;
}

.status-info {
  display: flex;
  gap: 16px;
  font-weight: bold;
}

.double-button {
  background-color: var(--color-gray);
}

.double-button.Red {
  background-color: var(--color-red);
}

.double-button.Blue {
  background-color: var(--color-blue);
}

.next-hole-button {
  width: 100%;
  padding: 12px;
  font-weight: bold;
} 