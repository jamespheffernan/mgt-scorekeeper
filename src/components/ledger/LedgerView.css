/* src/components/ledger/LedgerView.css */

/* 
  Renamed from 'bg-column' on the 'Big Game' table header cell 
  in the detailed view.
  Add desired styles here, for example, a background color.
*/
.ledger-big-game-header-cell {
  /* Example: background-color: #f3f4f6; /* A light grey, adjust as needed */
  /* Add other styling like text-align, padding, etc. if necessary */
} 