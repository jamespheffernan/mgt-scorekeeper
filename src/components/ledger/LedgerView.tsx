import { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useGameStore } from '../../store/gameStore';
import { Team } from '../../db/API-GameState';
import type { JunkEvent } from '../../store/gameStore';
import './LedgerView.css';
import { getFullName } from '../../utils/nameUtils';
import PlayerName from '../../components/PlayerName';

export const LedgerView = () => {
  const navigate = useNavigate();
  
  // Access store state
  const match = useGameStore(state => state.match);
  const players = useGameStore(state => state.players);
  const playerTeams = useGameStore(state => state.playerTeams);
  const ledger = useGameStore(state => state.ledger);
  const junkEvents = useGameStore(state => state.junkEvents);
  const bigGameRows = useGameStore(state => state.bigGameRows);
  const holeScores = useGameStore(state => state.holeScores);
  
  // Local state for drawer
  const [isOpen, setIsOpen] = useState(false);
  const [compactView, setCompactView] = useState(true);
  const [holeJunkEvents, setHoleJunkEvents] = useState<{ [hole: number]: JunkEvent[] }>({});
  
  // Group junk events by hole when junkEvents changes
  useEffect(() => {
    const groupedEvents: { [hole: number]: JunkEvent[] } = {};
    junkEvents.forEach(event => {
      if (!groupedEvents[event.hole]) {
        groupedEvents[event.hole] = [];
      }
      groupedEvents[event.hole].push(event);
    });
    setHoleJunkEvents(groupedEvents);
  }, [junkEvents]);
  
  // Toggle drawer
  const toggleDrawer = () => {
    setIsOpen(!isOpen);
  };
  
  // Toggle view mode
  const toggleViewMode = () => {
    setCompactView(!compactView);
  };
  
  // Get junk events for a specific hole
  const getJunkEventsForHole = (hole: number): JunkEvent[] => {
    return holeJunkEvents[hole] || [];
  };
  
  // Get junk total for a specific team on a specific hole
  const getTeamJunkTotalForHole = (hole: number, team: Team): number => {
    return getJunkEventsForHole(hole)
      .filter(event => event.teamId === team)
      .reduce((sum, event) => sum + event.value, 0);
  };
  
  // Get player name by ID
  const getPlayerName = (playerId: string): string => {
    const player = players.find(p => p.id === playerId);
    return player ? getFullName(player) : 'Unknown';
  };
  
  // Get team junk total
  const getTeamJunkTotal = (team: Team): number => {
    return junkEvents
      .filter(event => event.teamId === team)
      .reduce((sum, event) => sum + event.value, 0);
  };
  
  // Calculate team totals from running totals
  const getTeamTotals = () => {
    if (ledger.length === 0) return { red: 0, blue: 0 };
    
    const lastLedgerRow = ledger[ledger.length - 1];
    
    // Count the number of players on each team
    const redCount = playerTeams.filter(team => team === 'Red').length;
    const blueCount = playerTeams.filter(team => team === 'Blue').length;
    
    // Calculate team totals by dividing the sum by the number of team members
    // This ensures we display the correct team total, not the sum of all player amounts
    const redTotal = players.reduce((sum, _player, index) => 
      playerTeams[index] === 'Red' ? sum + lastLedgerRow.runningTotals[index] : sum, 0) / redCount;
    
    const blueTotal = players.reduce((sum, _player, index) => 
      playerTeams[index] === 'Blue' ? sum + lastLedgerRow.runningTotals[index] : sum, 0) / blueCount;
    
    console.log(`[DEBUG] Team totals (adjusted): Red=${redTotal}, Blue=${blueTotal}`);
    
    return { red: redTotal, blue: blueTotal };
  };
  
  // Determine winner of a hole
  const getHoleWinner = (holeIndex: number): 'Red' | 'Blue' | 'Push' => {
    if (!holeScores[holeIndex]) return 'Push';
    
    const { teamNet } = holeScores[holeIndex];
    if (teamNet[0] < teamNet[1]) return 'Red';
    if (teamNet[1] < teamNet[0]) return 'Blue';
    return 'Push';
  };
  
  // Format currency
  const formatCurrency = (amount: number): string => {
    const formattedAmount = Math.abs(amount).toString();
    const sign = amount >= 0 ? '+' : '-';
    return `${sign}$${formattedAmount}`;
  };
  
  // Export to CSV
  const exportToCsv = () => {
    // Create CSV header
    let csv = 'Hole,Base,Carry,Doubles,Payout';
    players.forEach(player => {
      csv += `,${getFullName(player)} Gross`;
    });
    players.forEach(player => {
      csv += `,${getFullName(player)} Net`;
    });
    players.forEach(player => {
      csv += `,${getFullName(player)} Money`;
    });
    csv += ',Red Junk,Blue Junk';
    if (match.bigGame) {
      csv += ',Big Game';
    }
    csv += '\n';
    
    // Add each ledger row
    ledger.forEach((row, index) => {
      const redJunk = getTeamJunkTotalForHole(row.hole, 'Red');
      const blueJunk = getTeamJunkTotalForHole(row.hole, 'Blue');
      csv += `${row.hole},${row.base},${row.carryAfter},${row.doubles},${row.payout + redJunk + blueJunk}`;
      // Add gross scores
      if (holeScores[index]) {
        holeScores[index].gross.forEach(gross => {
          csv += `,${gross}`;
        });
      } else {
        players.forEach(() => { csv += ','; });
      }
      // Add net scores
      if (holeScores[index]) {
        holeScores[index].net.forEach(net => {
          csv += `,${net}`;
        });
      } else {
        players.forEach(() => { csv += ','; });
      }
      // Add running totals
      row.runningTotals.forEach(total => {
        csv += `,${total}`;
      });
      // Add team junk totals
      csv += `,${getTeamJunkTotalForHole(row.hole, 'Red')}`;
      csv += `,${getTeamJunkTotalForHole(row.hole, 'Blue')}`;
      if (match.bigGame && bigGameRows[index]) {
        csv += `,${bigGameRows[index].subtotal}`;
      }
      csv += '\n';
    });
    
    // Add totals row
    csv += `Total,,,,`;
    if (ledger.length > 0) {
      const lastRow = ledger[ledger.length - 1];
      players.forEach(() => { csv += ','; }); // For gross
      players.forEach(() => { csv += ','; }); // For net
      lastRow.runningTotals.forEach(total => {
        csv += `,${total}`;
      });
      // Add total team junk
      csv += `,${getTeamJunkTotal('Red')}`;
      csv += `,${getTeamJunkTotal('Blue')}`;
    }
    if (match.bigGame) {
      csv += `,${match.bigGameTotal}`;
    }
    
    // Trigger download
    const blob = new Blob([csv], { type: 'text/csv' });
    const url = URL.createObjectURL(blob);
    const a = document.createElement('a');
    a.setAttribute('hidden', '');
    a.setAttribute('href', url);
    a.setAttribute('download', `millbrook-game-${match.date}.csv`);
    document.body.appendChild(a);
    a.click();
    document.body.removeChild(a);
  };
  
  // Calculate team colors for display
  const teamColor = (team: Team) => {
    return team === 'Red' ? 'team-red' : 'team-blue';
  };
  
  // Get class for winner
  const getWinnerClass = (winner: 'Red' | 'Blue' | 'Push') => {
    if (winner === 'Push') return 'result-push';
    return winner === 'Red' ? 'result-red' : 'result-blue';
  };
  
  // Team totals for display
  const teamTotals = getTeamTotals();
  
  // Determine if we have any data to display
  const hasData = ledger.length > 0;
  
  // Calculate running Big Game total (enhanced for T-13)
  const getRunningBigGameTotal = (upToIndex: number): number => {
    return bigGameRows
      .slice(0, upToIndex + 1)
      .reduce((total, row) => total + row.subtotal, 0);
  };

  // Function to navigate back to current hole
  const handleBackToHole = () => {
    navigate(`/hole/${match.currentHole}`);
  };

  return (
    <div className="ledger-container">
      <div className="back-button-container">
        <button 
          className="back-to-hole-button"
          onClick={handleBackToHole}
        >
          Back to Hole {match.currentHole}
        </button>
      </div>
      
      <div className="ledger-header" onClick={toggleDrawer}>
        <h3>Running Totals</h3>
        <div className="team-totals">
          <span className="team-red">Red {formatCurrency(teamTotals.red)}</span>
          <span className="team-blue">Blue {formatCurrency(teamTotals.blue)}</span>
          {match.bigGame && (
            <span className="big-game-total">Big Game: {match.bigGameTotal}</span>
          )}
        </div>
        <div className="drawer-toggle">{isOpen ? '▲' : '▼'}</div>
      </div>
      
      {isOpen && (
        <div className="ledger-details">
          {hasData ? (
            <>
              <div className="ledger-controls">
                <div className="view-controls">
                  <button 
                    className="view-toggle-button"
                    onClick={toggleViewMode}
                  >
                    {compactView ? 'Show Detailed View' : 'Show Compact View'}
                  </button>
                  <button 
                    className="export-button"
                    onClick={exportToCsv}
                  >
                    Export CSV
                  </button>
                </div>
              </div>
              
              {compactView ? (
                <table className="ledger-table compact-view">
                  <thead>
                    <tr>
                      <th>Hole</th>
                      <th>Base</th>
                      <th>Carry</th>
                      <th>Dbl</th>
                      <th>Result</th>
                      <th>Win $</th>
                      <th>Red Junk</th>
                      <th>Blue Junk</th>
                      {match.bigGame && <th>Big Game</th>}
                      {match.bigGame && <th>Running</th>}
                    </tr>
                  </thead>
                  <tbody>
                    {ledger.map((row, index) => {
                      const holeWinner = getHoleWinner(index);
                      const redJunk = getTeamJunkTotalForHole(row.hole, 'Red');
                      const blueJunk = getTeamJunkTotalForHole(row.hole, 'Blue');
                      const bgRow = bigGameRows[index];
                      const runningBigGameTotal = getRunningBigGameTotal(index);
                      
                      return (
                        <tr key={index} className={index === match.currentHole - 1 ? "current-hole-row" : ""}>
                          <td>{row.hole}</td>
                          <td>${row.base}</td>
                          <td>${row.carryAfter}</td>
                          <td>{row.doubles > 0 ? '✓' : ''}</td>
                          <td className={getWinnerClass(holeWinner)}>{holeWinner}</td>
                          <td className="win-amount">${row.payout + redJunk + blueJunk}</td>
                          <td className="junk-amount team-red">${redJunk}</td>
                          <td className="junk-amount team-blue">${blueJunk}</td>
                          {match.bigGame && (
                            <td className="big-game-cell">
                              {bgRow ? bgRow.subtotal : '-'}
                            </td>
                          )}
                          {match.bigGame && (
                            <td className="big-game-running">
                              {runningBigGameTotal}
                            </td>
                          )}
                        </tr>
                      );
                    })}
                  </tbody>
                  <tfoot>
                    <tr>
                      <td colSpan={2}>Final Totals</td>
                      <td colSpan={match.bigGame ? 8 : 6}>
                        <div className="player-totals-grid">
                          {players.map((player, index) => (
                            <div 
                              key={index} 
                              className={`player-total ${teamColor(playerTeams[index])}`}
                            >
                              {getFullName(player)}: {formatCurrency(ledger.length > 0 ? ledger[ledger.length - 1].runningTotals[index] : 0)}
                            </div>
                          ))}
                          <div className="team-junk-totals">
                            <div className="team-red">Red Junk: ${getTeamJunkTotal('Red')}</div>
                            <div className="team-blue">Blue Junk: ${getTeamJunkTotal('Blue')}</div>
                          </div>
                        </div>
                        {match.bigGame && (
                          <div className="big-game-final">
                            Big Game Total: {match.bigGameTotal}
                          </div>
                        )}
                      </td>
                    </tr>
                  </tfoot>
                </table>
              ) : (
                <table className="ledger-table detailed-view">
                  <thead>
                    <tr>
                      <th>Hole</th>
                      <th>Base</th>
                      <th>Carry</th>
                      <th>Dbl</th>
                      <th>Result</th>
                      {players.map((_, playerIndex) => (
                        <th key={playerIndex} className={teamColor(playerTeams[playerIndex])}>
                          {getFullName(players[playerIndex])}
                        </th>
                      ))}
                      {match.bigGame && <th className="ledger-big-game-header-cell">Big Game</th>}
                    </tr>
                  </thead>
                  <tbody>
                    {ledger.map((row, index) => {
                      const holeWinner = getHoleWinner(index);
                      const bgRow = bigGameRows[index];
                      
                      return (
                        <tr key={index} className={index === match.currentHole - 1 ? "current-hole-row" : ""}>
                          <td>{row.hole}</td>
                          <td>${row.base}</td>
                          <td>${row.carryAfter}</td>
                          <td>{row.doubles > 0 ? '✓' : ''}</td>
                          <td className={getWinnerClass(holeWinner)}>{holeWinner}</td>
                          
                          {players.map((_, playerIndex) => {
                            const teamJunkEvents = getJunkEventsForHole(row.hole)
                              .filter(event => event.teamId === playerTeams[playerIndex]);
                            const teamJunkValue = teamJunkEvents.reduce((sum, event) => sum + event.value, 0);
                            const runningTotal = row.runningTotals[playerIndex];
                            const previousTotal = index > 0 ? ledger[index - 1].runningTotals[playerIndex] : 0;
                            const holeChange = runningTotal - previousTotal;
                            
                            return (
                              <td key={playerIndex} className="player-ledger-cell">
                                <div className={`player-hole-change ${holeChange > 0 ? 'positive' : holeChange < 0 ? 'negative' : ''}`}>
                                  {formatCurrency(holeChange)}
                                </div>
                                {teamJunkValue > 0 && (
                                  <div className="player-junk-events">
                                    {teamJunkEvents.map((event, eventIndex) => (
                                      <div key={eventIndex} className={`junk-pill junk-${event.type.toLowerCase()}`}>
                                        {event.type} ${event.value} ({getPlayerName(event.playerId)})
                                      </div>
                                    ))}
                                  </div>
                                )}
                                <div className="player-running-total">
                                  Total: {formatCurrency(runningTotal)}
                                </div>
                              </td>
                            );
                          })}
                          
                          {match.bigGame && (
                            <td className="big-game-cell">
                              <div className="big-game-detail">
                                <div className="big-game-scores">
                                  <span className="best-net">{bgRow?.bestNet[0]}</span>
                                  <span className="best-net-plus">+</span>
                                  <span className="best-net">{bgRow?.bestNet[1]}</span>
                                </div>
                                <div className="big-game-subtotal">= {bgRow?.subtotal}</div>
                              </div>
                              <div className="big-game-running">
                                Running: {getRunningBigGameTotal(index)}
                              </div>
                            </td>
                          )}
                        </tr>
                      );
                    })}
                  </tbody>
                  <tfoot>
                    <tr>
                      <td colSpan={5}>
                        <strong>Final Totals</strong>
                      </td>
                      
                      {players.map((_, index) => {
                        const finalTotal = ledger.length > 0 ? ledger[ledger.length - 1].runningTotals[index] : 0;
                        
                        return (
                          <td key={index} className={`player-final-cell ${teamColor(playerTeams[index])}`}>
                            <div className="player-final-total">
                              {formatCurrency(finalTotal)}
                            </div>
                          </td>
                        );
                      })}
                      
                      {match.bigGame && (
                        <td className="big-game-final-cell">
                          <strong>{match.bigGameTotal}</strong>
                          <div className="big-game-detail-link">
                            Per hole breakdown above
                          </div>
                        </td>
                      )}
                    </tr>
                  </tfoot>
                </table>
              )}
            </>
          ) : (
            <div className="ledger-empty">
              <p>No score data available yet. Complete a hole to see the ledger.</p>
            </div>
          )}
        </div>
      )}
    </div>
  );
}; 