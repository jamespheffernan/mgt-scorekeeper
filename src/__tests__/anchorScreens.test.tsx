import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { jest } from '@jest/globals';

// Stable mock players array to avoid new reference on every render
const stableMockPlayers: any[] = [];

// Mock Firestore players hook
jest.mock('../hooks/useFirestorePlayers', () => ({
  useFirestorePlayers: () => ({
    players: stableMockPlayers,
    isLoading: false,
    createPlayer: jest.fn(),
    updatePlayer: jest.fn(),
    deletePlayerById: jest.fn(),
    error: null,
  }),
}));

// Mock Auth context
jest.mock('../context/AuthContext', () => ({
  useAuth: () => ({
    currentUser: null,
    loading: false,
    error: null,
    signUp: async () => null,
    logIn: async () => null,
    logOut: async () => {},
    authInitialized: true,
  }),
}));

import MatchSetup from '../components/setup/MatchSetup';
import { HoleView } from '../components/hole/HoleView';
import { LedgerView } from '../components/ledger/LedgerView';
import SettlementView from '../components/ledger/SettlementView';

// Mock scrollTo for jsdom environment
beforeAll(() => {
  Object.defineProperty(HTMLElement.prototype, 'scrollTo', {
    configurable: true,
    value: jest.fn(),
  });
});

describe('Snapshot tests for anchor screens', () => {
  it('renders MatchSetup correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <MatchSetup />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders HoleView correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <HoleView />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders LedgerView correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <LedgerView />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders SettlementView correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SettlementView matchId="test" />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
}); 