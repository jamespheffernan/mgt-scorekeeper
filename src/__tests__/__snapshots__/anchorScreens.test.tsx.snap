// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for anchor screens renders HoleView correctly 1`] = `
<DocumentFragment>
  <div
    class="hole-view"
  >
    <div
      class="hole-header"
    >
      <h2>
        Hole 1
      </h2>
      <div
        class="hole-actions"
      >
        <button
          aria-label="View full ledger"
          class="view-ledger-button"
        >
          <span
            class="button-icon"
          >
            ðŸ“Š
          </span>
           View Ledger
        </button>
        <button
          aria-label="Cancel the game"
          class="cancel-game-button"
        >
          <span
            class="button-icon"
          >
            â›”
          </span>
           Cancel Game
        </button>
      </div>
    </div>
    <div
      class="scores-container"
    >
      <h3>
        Enter Scores
      </h3>
    </div>
    <div
      class="hole-controls"
    >
      <button
        aria-label="Submit scores and advance to next hole"
        class="submit-button"
      >
        Submit Scores
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Snapshot tests for anchor screens renders LedgerView correctly 1`] = `
<DocumentFragment>
  <div
    class="ledger-container"
  >
    <div
      class="back-button-container"
    >
      <button
        class="back-to-hole-button"
      >
        Back to Hole 1
      </button>
    </div>
    <div
      class="ledger-header"
    >
      <h3>
        Running Totals
      </h3>
      <div
        class="team-totals"
      >
        <span
          class="team-red"
        >
          Red +$0
        </span>
        <span
          class="team-blue"
        >
          Blue +$0
        </span>
      </div>
      <div
        class="drawer-toggle"
      >
        â–¼
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Snapshot tests for anchor screens renders MatchSetup correctly 1`] = `
<DocumentFragment>
  <header
    class="top-bar-root safe-wrapper"
    style="display: flex; justify-content: center; align-items: center;"
  >
    <h1
      class="topbar-title"
      style="text-align: center;"
    >
      Game Setup
    </h1>
  </header>
  <div
    class="setup-container mobile-setup-container"
  >
    <div
      class="stepper-root "
    >
      <h2
        class="stepper-title"
      >
        Step 1 of 2
      </h2>
      <div
        class="stepper-bar-container"
      >
        <div
          class="stepper-bar-item stepper-bar-item-active"
        />
        <div
          class="stepper-bar-item stepper-bar-item-inactive"
        />
      </div>
    </div>
    <div
      class="player-roster-container mobile-player-roster"
    >
      <div
        style="background: rgb(255, 224, 224); padding: 8px; margin: 8px 0px; font-size: 12px; border: 1px solid #ff0000; flex-shrink: 0;"
      >
        <h4>
          Debug Mode
        </h4>
        <button
          style="font-size: 12px; padding: 2px 5px;"
        >
          Hide Debug
        </button>
        <div>
          <strong>
            Players data:
          </strong>
          <pre
            style="max-height: 100px; overflow: auto; font-size: 10px;"
          >
            []
          </pre>
        </div>
        <div>
          Check browser console for complete sorting logs
        </div>
      </div>
      <div
        class="selected-players-summary mobile-selected-players-sticky"
      >
        <h4
          class="selected-players-title"
        >
          Selected Players
        </h4>
        <div
          class="selected-players-grid"
        >
          <div
            class="empty-player-slot empty-player-slot-red"
            style="background-color: rgba(255, 0, 0, 0.1);"
          >
            Select Red Player
          </div>
          <div
            class="empty-player-slot empty-player-slot-blue"
            style="background-color: rgba(0, 0, 255, 0.1);"
          >
            Select Blue Player
          </div>
          <div
            class="empty-player-slot empty-player-slot-red"
            style="background-color: rgba(255, 0, 0, 0.1);"
          >
            Select Red Player
          </div>
          <div
            class="empty-player-slot empty-player-slot-blue"
            style="background-color: rgba(0, 0, 255, 0.1);"
          >
            Select Blue Player
          </div>
        </div>
      </div>
      <div
        class="search-box-wrapper"
      >
         
        <input
          class="search-input-field"
          placeholder="Search Players..."
          type="text"
          value=""
        />
      </div>
      <button
        class="add-player-button btn-primary general-button-styling"
      >
        Add New Player
      </button>
      <div
        class="player-lists-container mobile-scrollable-list-container"
      >
        <p
          class="no-players-message mobile-centered-text"
        >
          No players available. Try adding some!
        </p>
      </div>
      <div
        class="confirm-selection-section mobile-confirm-button-sticky"
      >
        <button
          aria-live="polite"
          class="confirm-button mobile-button-fullwidth primary-action-button"
          disabled=""
        >
          Select 4 More Player(s)
        </button>
      </div>
    </div>
    <div
      class="setup-actions mobile-full-width-buttons"
    >
      <button
        class="secondary-action-button"
      >
        View Course Details
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Snapshot tests for anchor screens renders SettlementView correctly 1`] = `
<DocumentFragment>
  <div
    class="settlement-view"
  >
    <div
      class="settlement-header"
    >
      <h2>
        Game Summary
      </h2>
      <div
        class="settlement-date"
      >
        Invalid Date
      </div>
    </div>
    <div
      class="tab-container"
    >
      <div
        class="settlement-tabs"
      >
        <button
          class="tab-button active"
          data-tab-id="summary"
        >
          Summary
        </button>
        <button
          class="tab-button "
          data-tab-id="details"
        >
          Hole-by-Hole
        </button>
      </div>
      <div
        class="settlement-content"
      >
        <div
          class="settlement-summary-tab"
        >
          <div
            class="game-result-banner"
          >
            <h3>
              It's a Tie!
            </h3>
          </div>
          <div
            class="team-summary"
          >
            <div
              class="team-total team-red "
            >
              <div
                class="team-name"
              >
                Red Team
              </div>
              <div
                class="team-amount"
              >
                +$0
              </div>
              <div
                class="team-holes-won"
              >
                0 holes won
              </div>
            </div>
            <div
              class="team-summary-separator"
            >
              vs
            </div>
            <div
              class="team-total team-blue "
            >
              <div
                class="team-name"
              >
                Blue Team
              </div>
              <div
                class="team-amount"
              >
                +$0
              </div>
              <div
                class="team-holes-won"
              >
                0 holes won
              </div>
            </div>
          </div>
          <div
            class="player-results"
          />
          <div
            class="game-stats"
          >
            <div
              class="stats-row"
            >
              <div
                class="stat-item"
              >
                <div
                  class="stat-label"
                >
                  Holes Played
                </div>
                <div
                  class="stat-value"
                >
                  0
                </div>
              </div>
              <div
                class="stat-item"
              >
                <div
                  class="stat-label"
                >
                  Duration
                </div>
                <div
                  class="stat-value"
                >
                  00:00
                </div>
              </div>
              <div
                class="stat-item"
              >
                <div
                  class="stat-label"
                >
                  Doubles
                </div>
                <div
                  class="stat-value"
                >
                  0
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="settlement-actions"
    >
      <div
        class="export-actions"
      >
        <button
          class="export-button"
        >
          Export CSV
        </button>
        <button
          class="export-button"
        >
          Save PNG
        </button>
      </div>
      <div
        class="game-actions"
      >
        <button
          class="action-button rematch-button"
        >
          Rematch
        </button>
        <button
          class="action-button new-game-button"
        >
          New Game
        </button>
        <button
          class="action-button menu-button"
        >
          Main Menu
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
